server/src/test/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpointTests.java

```diff
@@ -1201,7 +1255,7 @@ public void testLoginHintOriginOidcForJson() throws Exception {
         assertNotNull(model.get("prompts"));
         assertTrue(model.get("prompts") instanceof Map);
         Map<String, String[]> returnedPrompts = (Map<String, String[]>)model.get("prompts");
-        assertEquals(3, returnedPrompts.size());
+        assertEquals(2, returnedPrompts.size());
```

Comment:
why returnedPrompts.size is 2 here?

Reply:
`returnedPrompts` uses `samlIdentityProviders` and `oauthIdentityProviders` lists to add prompts to the model. Before we would always fetch all idps which would add all prompts (in old test the three returned are `passcode`, `username`, and `password`). With this change externalIdps aren't fetched so `passcode` prompt is excluded which changed the assertion from 3 to 2.

For reference the logic to exclude passcode when no idps are configured happens [here](https://github.com/GESoftware-CF/uaa/blob/ff9e8d5041d9ec708df0f07c5bd1cbca56ca73ee/server/src/main/java/org/cloudfoundry/identity/uaa/login/LoginInfoEndpoint.java#L756)